bin/tx-prover/build.rs

```diff
@@ -0,0 +1,96 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+};
+
+use miette::IntoDiagnostic;
+use protox::prost::Message;
+
+/// Generates Rust protobuf bindings from .proto files in the root directory.
+///
+/// This is done only if BUILD_PROTO environment variable is set to `1` to avoid running the script
+/// on crates.io where repo-level .proto files are not available.
+fn main() -> miette::Result<()> {
+    println!("cargo::rerun-if-changed=../../proto");
+    println!("cargo::rerun-if-env-changed=BUILD_PROTO");
+
+    // Skip this build script in BUILD_PROTO environment variable is not set to `1`.
+    if env::var("BUILD_PROTO").unwrap_or("0".to_string()) == "0" {
+        return Ok(());
+    }
```

Comment:
Since the `.proto` file is now a part of the crate, I don't think need to worry about compilation on crates.io. So, `BUILD_PROTO` environment variable may not be needed.